lib/portlayer/exec/handle.go

```diff
@@ -147,6 +147,19 @@ func (h *Handle) Commit(ctx context.Context, sess *session.Session, waitTime *in
 	// make sure there is a spec
 	h.SetSpec(nil)
 	cfg := make(map[string]string)
+
+	// Set timestamps based on target state
+	switch *h.State {
```

Comment:
Yes -- they won't be exact...but the time difference should be minimal and acceptable.